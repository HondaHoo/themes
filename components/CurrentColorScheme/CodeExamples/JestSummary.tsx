import {
  Foreground,
  Background,
  Red,
  Green,
  Yellow,
} from '@/components/CurrentColorScheme/CodeExamples/ColorComponents';

const Gaps = ({count}: {count: number}) => <>{Array(count).fill(' ')}</>;

const JestSummary = () => {
  return (
    <figure>
      <figcaption>Jest summary example</figcaption>
      <div>
        limit50x----------------------------------------- Test Suites: 1 failed,
        9 passed, 10 total Tests: 1 failed, 36 passed, 37 total Snapshots: 1
        passed, 1 total Time: 2.897 s Ran all test suites.
      </div>
      <Background>
        <div>
          <Green bold inverse>
            {' '}
            PASS{' '}
          </Green>{' '}
          <Foreground dim>components/ToggleCurrentLightness/</Foreground>
          <Foreground bold>ToggleCurrentLightness.test.tsx</Foreground>
        </div>
        <div>
          <Red bold inverse>
            {' '}
            FAIL{' '}
          </Red>{' '}
          <Foreground dim>components/NextPrevButton/</Foreground>
          <Foreground bold>NextPrevButton.test.tsx</Foreground>
        </div>
        <div>
          <Foreground bold>Test Suites: </Foreground>
          <Red bold>1 failed</Red>, <Green bold>9 passed</Green>, 10 total
        </div>
        <div>
          <Foreground bold>
            Time: <Gaps count={6} />
          </Foreground>{' '}
          <Yellow bold>2.897</Yellow>
        </div>
        <div>
          <Foreground dim>Ran all test suites.</Foreground>
        </div>
      </Background>
    </figure>
  );
};

export default JestSummary;
