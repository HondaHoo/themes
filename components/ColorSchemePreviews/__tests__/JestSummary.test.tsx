import '@testing-library/jest-dom';
import {render, screen} from '@testing-library/react';

import JestSummary from '../JestPreview';

it('should render all the components', () => {
  render(<JestSummary />);
  expect(screen.getByRole('figure')).toMatchSnapshot();
});
