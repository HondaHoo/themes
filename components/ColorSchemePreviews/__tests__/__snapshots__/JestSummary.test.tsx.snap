// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render all the components 1`] = `
<figure
  class="container"
>
  <figcaption>
    <span
      style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
    >
      Terminal code styled in the active color themes
    </span>
  </figcaption>
  <div
    class="background"
  >
    <div>
      <span
        style="font-weight: 700;"
      >
         
        PASS
         
      </span>
       
      <span
        style="opacity: 0.5;"
      >
        components/ToggleLightness/
      </span>
      <span
        style="font-weight: 700;"
      >
        ToggleLightness.test.tsx
      </span>
    </div>
    <div>
      <span
        style="font-weight: 700;"
      >
         
        FAIL
         
      </span>
       
      <span
        style="opacity: 0.5;"
      >
        components/NextPrevButton/
      </span>
      <span
        style="font-weight: 700;"
      >
        NextPrevButton.test.tsx
      </span>
    </div>
    <div>
      <span
        style="font-weight: 700;"
      >
        Test Suites: 
      </span>
      <span
        style="font-weight: 700;"
      >
        1 failed
      </span>
      , 
      <span
        style="font-weight: 700;"
      >
        9 passed
      </span>
      , 10 total
    </div>
    <div>
      <span
        style="font-weight: 700;"
      >
        Time: 
         
         
         
         
         
         
      </span>
       
      <span
        style="font-weight: 700;"
      >
        2.897
      </span>
    </div>
    <div>
      <span
        style="opacity: 0.5;"
      >
        Ran all test suites.
      </span>
    </div>
  </div>
</figure>
`;
